### PR TITLE
Multiple routing strategies support 

### PR DESCRIPTION
**Pull request by umpirsky**
Fixes issue #1.

- [x] Remove factories and introuce strategy
- [x] Implement `prefix_except_default` strategy.
- [x] Override router for `prefix_except_default` strategy.

### CODE REVIEW

**Comment by kbond** [comment]
This is looking good, I've been looking for a simplified and maintained alternative to the jms bundle.

For overriding the router, you are extending the routing component router but shouldn't you extend the frameworkbundle router? That one is warmable.

File: src/Routing/Router.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Umpirsky\I18nRoutingBundle\Routing;
+
+use Symfony\Component\Routing\Router as SymfonyRouter;
+
+class Router extends SymfonyRouter
```
Comment (kbond) [inline src/Routing/Router.php]: What is the reason for overriding the router here?

Comment (umpirsky) [reply src/Routing/Router.php]: @kbond When using `prefix_except_default` strategy, and you are on `/es/route66` url, you want all urls on the page to be `/es` prefixed.

**Comment by umpirsky** [comment]
> For overriding the router, you are extending the routing component router but shouldn't you extend the frameworkbundle router? That one is warmable.

@kbond Yes, the warmable one, that's why tests were failing. Fixed in my last commit. :)

This is still WIP.

**Comment by kbond** [comment]
I know this is a WIP but I just tried hot-swapping the jms bundle with this one on a project - here are some notes:

1. the `_i18n` route should come before the original. I have a `/{path}` route that acts as a catch all with a `.+` requirement for `{path}` - because the `i18n` version comes after it never gets hit and the i18n version of the urls give 404s.
2. the `_i18n` route should not have the default locale in the requirements `/en/path` should not be available if the default locale is en
3. previously my homepage was accessibly via `/` and `/fr/` - now it's `/` and `/fr` (note the missing trailing slash now) - I am now getting a 404 on `/fr/` (previously I would get a 404 on `/fr`...)

**Comment by umpirsky** [comment]
@kbond Maybe it's worth opening each of this as a separate issue, it's easier to discuss them separately. 

1. While this is easy to fix on my end, not sure if it's a right thing to do, I' afraid of other edge cases that can appear. It's easy to fix on your end by adding `i18n: false` option to your route and manually creating i18n route for it.
1. What strategy `prefix_except_default` or `prefix`?
1. That must be the difference between jms and this bundle. Why is trailing slash preferred?

Thanks for the feedback, really appreciate it.

**Comment by kbond** [comment]
None of these are really bugs - just differences with jms

1. Fair enough, I opened an issue #5 to discuss further
2. See #6
3. See #7

I can open some PRs too  - I just wanted to wait until it was no longer WIP

**Comment by umpirsky** [comment]
@kbond Sure, contributions are encouraged, thanks!

To prevent overlapping, just open issue and ping me if you start working on something. I'll be there for any questions, help, etc...

**Pull request merged**